### PR TITLE
Privacy plugin for privacy tools in Joomla 3.9 #25

### PR DESCRIPTION
**Pull request by Mayaank12**

### CODE REVIEW

File: src/plugins/privacy/activitystream/activitystream.php [hunk 1]
```diff
@@ -0,0 +1,185 @@
...
+	protected $db;
+
+	/**
+	 * Reports the privacy related capabilities for this plugin to site administrators.
+	 *
+	 * @return  array
+	 *
+	 * @since   __DEPLOY_VERSION__
+	 */
+	public function onPrivacyCollectAdminCapabilities()
```
Comment (ankush-maherwal) [inline src/plugins/privacy/activitystream/activitystream.php]: @Mayaank12 
don't show capabilities for horizontal extensions

File: src/plugins/privacy/activitystream/activitystream.php [hunk 2]
```diff
@@ -0,0 +1,185 @@
...
+	 * @return  PrivacyExportDomain
+	 *
+	 * @since   __DEPLOY_VERSION__
+	 */
+	private function createActivityStreamDomain(JTableUser $user)
+	{
+		$domain = $this->createDomain('ActivityStream', 'ActivityStream data');
+
+		$query = $this->db->getQuery(true)
+			->select('actor_id, object_id, target_id, type, template')
```
Comment (ankush-maherwal) [inline src/plugins/privacy/activitystream/activitystream.php]: @Mayaank12 
use quoteName()

File: src/plugins/privacy/activitystream/activitystream.php [hunk 3]
```diff
@@ -0,0 +1,167 @@
...
+		{
+			return;
+		}
+
+		$db = $this->db;
+
+		// Delete Activity Stream user data :
+		$query = $db->getQuery(true)
+					->delete($db->quoteName('#__tj_activities'))
+					->where('actor_id = ' . $user->id);
```
Comment (ankush-maherwal) [inline src/plugins/privacy/activitystream/activitystream.php]: @Mayaank12 
use query builder

File: src/plugins/privacy/activitystream/activitystream.xml [hunk 4]
```diff
@@ -0,0 +1,25 @@
+<?xml version="1.0" encoding="utf-8"?>
+<extension version="3.9" type="plugin" group="privacy" method="upgrade">
+	<name>plg_privacy_activitystream</name>
+	<version>__DEPLOY_VERSION__</version>
+	<creationDate>5th October 2018</creationDate>
```
Comment (ankush-maherwal) [inline src/plugins/privacy/activitystream/activitystream.xml]: @Mayaank12 
Use 
__RELEASE_DATE__
for creation date

File: src/plugins/privacy/activitystream/activitystream.php [hunk 5]
```diff
@@ -0,0 +1,167 @@
...
+	 * @return  PrivacyExportDomain
+	 *
+	 * @since   __DEPLOY_VERSION__
+	 */
+	private function createActivityStreamDomain(JTableUser $user)
+	{
+		$domain = $this->createDomain('ActivityStream', 'ActivityStream data');
+		$db = Factory::getDbo();
+		$query = $db->getQuery(true)
+			->select($db->quote(array('id', 'actor_id', 'object_id', 'target_id', 'type', 'template')))
```
Comment (ankush-maherwal) [inline src/plugins/privacy/activitystream/activitystream.php]: @Mayaank12 
$db->quoteName(array('id'))

File: src/plugins/privacy/activitystream/activitystream.php (hunk 3 above)
Comment (Mayaank12) [reply src/plugins/privacy/activitystream/activitystream.php]: Done

File: src/plugins/privacy/activitystream/activitystream.xml (hunk 4 above)
Comment (Mayaank12) [reply src/plugins/privacy/activitystream/activitystream.xml]: Done

File: src/plugins/privacy/activitystream/activitystream.php (hunk 5 above)
Comment (Mayaank12) [reply src/plugins/privacy/activitystream/activitystream.php]: Done

**Pull request merged**